hydra-cluster/test/Test/DirectChainSpec.hs

```diff
@@ -383,15 +385,26 @@ withDirectChainTest tracer config ctx action = do
         , waitCallback = atomically $ takeTMVar eventMVar
         }
 
+hasInitTxWith :: (HasCallStack, IsTx tx) => ContestationPeriod -> [Party] -> OnChainTx tx -> Expectation
+hasInitTxWith expectedContestationPeriod expectedParties = \case
+  OnInitTx{contestationPeriod, parties} -> do
+    expectedContestationPeriod `shouldBe` contestationPeriod
+    expectedParties `shouldBe` parties
+  tx -> failure ("Unexpected observation: " <> show tx)
+
 observesInTime :: IsTx tx => DirectChainTest tx IO -> OnChainTx tx -> IO ()
-observesInTime DirectChainTest{waitCallback} expected =
+observesInTime chain expected =
+  observesInTimeSatisfying chain (`shouldBe` expected)
+
+observesInTimeSatisfying :: DirectChainTest tx IO -> (OnChainTx tx -> Expectation) -> IO ()
+observesInTimeSatisfying DirectChainTest{waitCallback} check =
   failAfter 10 go
  where
   go = do
     e <- waitCallback
     case e of
       Observation{observedTx} -> do
```

Comment:
Could: remove redundant do

Reply:
I vaguely remember someone (@KtorZ ?) saying these were more useful than harmful even for single line monadic actions, but I personally tend to agree